backends/kv/writer.go

```diff
@@ -428,14 +414,12 @@ func validColName(name string) string {
 
 // WaitForComplete waits for write to complete
 func (a *Appender) WaitForComplete(timeout time.Duration) error {
-	a.logger.DebugWith("WaitForComplete", "sent", a.sent)
-	a.commChan <- a.sent
-	<-a.doneChan
+	a.Close()
```

Comment:
waitForComplete should wait for all items to be sent, this just closes the channel and returns but writes still may happen after that, and you can get errors as well.